omnigibson/tasks/grasp_task.py

```diff
@@ -0,0 +1,140 @@
+import random
+import numpy as np
+import omnigibson as og
+from omnigibson.action_primitives.starter_semantic_action_primitives import UndoableContext
+from omnigibson.reward_functions.grasp_reward import GraspReward
+
+from omnigibson.tasks.task_base import BaseTask
+from omnigibson.scenes.scene_base import Scene
+from omnigibson.termination_conditions.grasp_goal import GraspGoal
+from omnigibson.termination_conditions.timeout import Timeout
+from omnigibson.utils.grasping_planning_utils import get_grasp_poses_for_object_sticky
+from omnigibson.utils.motion_planning_utils import set_arm_and_detect_collision
+from omnigibson.utils.python_utils import classproperty
+from omnigibson.utils.sim_utils import land_object
+
+DIST_COEFF = 0.1
+GRASP_REWARD = 1.0
+MAX_JOINT_RANDOMIZATION_ATTEMPTS = 50
+
+class GraspTask(BaseTask):
+    """
+    Grasp task
+    """
+
+    def __init__(
+        self,
+        obj_name,
+        termination_config=None,
+        reward_config=None,
+    ):
+        self.obj_name = obj_name
+        super().__init__(termination_config=termination_config, reward_config=reward_config)
+        
+    def _load(self, env):
+        pass
+
+    def _create_termination_conditions(self):
+        terminations = dict()
+        terminations["graspgoal"] = GraspGoal(
+            self.obj_name
+        )
+        terminations["timeout"] = Timeout(max_steps=self._termination_config["max_steps"])
+
+        return terminations
+
+    def _create_reward_functions(self):
+        rewards = dict()
+        rewards["grasp"] = GraspReward(
+            self.obj_name,
+            dist_coeff=self._reward_config["r_dist_coeff"],
+            grasp_reward=self._reward_config["r_grasp"]
+        )
+        return rewards
+    
+    def _reset_agent(self, env):
+        # from IPython import embed; embed()
+        if hasattr(env, '_primitive_controller'):
```

Comment:
this seems incredible specific to the primitives code. Why is it specific to the primitive controller? Shouldn't the randomization occur anyways?